### PR TITLE
fix(datetime): make sure mergeDateTime returns local time

### PR DESCRIPTION
**Pull request by petrjasek**
and not utc.

also moved `datetimeHelper` service to `superdesk.datetime` module from `superdesk.ui`

### CODE REVIEW

**Comment by ioanpocol** [comment]
:+1:

File: scripts/superdesk/datetime/datetime.js [hunk 1]
```diff
@@ -73,6 +73,42 @@
...
+            return moment(value, dateFormat, true).isValid();
+        };
+
+        this.mergeDateTime = function(date_str, time_str, timezone) {
+            var tz = timezone || config.defaultTimezone;
+            var merge_str = date_str + ' ' + time_str;
+            var formatter = config.model.dateformat + ' ' + config.model.timeformat;
+
+            // return without timezone information, which is stored separately
+            return moment.tz(merge_str, formatter, tz).format('YYYY-MM-DD[T]HH:mm:ss');
```
Comment (petrjasek) [inline scripts/superdesk/datetime/datetime.js]: fyi calling `.format`  here is the only change, rest is just moving the code around

**Comment by akintolga** [comment]
:+1:

**Comment by mdhaman** [comment]
:+1:

File: scripts/superdesk/datetime/datetime.js [hunk 2]
```diff
@@ -73,6 +73,42 @@
...
+
+        /*
+        * @param timestring 2016-03-01T04:45:00+0000
+        * @param timezone Europe/London
+        */
+        this.splitDateTime = function(timestring, timezone) {
+            var momentTS = moment.tz(timestring, timezone);
+
+            return {
+                'date': momentTS.format(config.model.dateformat),
```
Comment (mdhaman) [inline scripts/superdesk/datetime/datetime.js]: @petrjasek from [Line 190](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/superdesk-authoring/metadata/metadata.js#L190) the `timestring` passed is `2016-04-20T09:45:00+0000` i.e. it contains utcoffset. This causes moment to calculate the time part wrong.

Comment (petrjasek) [inline scripts/superdesk/datetime/datetime.js]: that's what is changed in PR, the time won't be utc anymore, but it will be local according to timezone

**Pull request merged**